app/instrumentation-tests/src/org/commcare/androidTests/DateWidgetsTests.kt

```diff
@@ -0,0 +1,171 @@
+import androidx.test.espresso.action.ViewActions
+import androidx.test.espresso.contrib.PickerActions
+import androidx.test.espresso.matcher.ViewMatchers
+import androidx.test.espresso.matcher.ViewMatchers.*
+import androidx.test.ext.junit.runners.AndroidJUnit4
+import androidx.test.filters.LargeTest
+import org.commcare.annotations.BrowserstackTests
+import org.commcare.dalvik.R
+import org.commcare.utils.InstrumentationUtility
+import org.commcare.utils.isPresent
+import org.hamcrest.Matchers
+import org.hamcrest.Matchers.allOf
+import org.junit.Assert.*
+import org.junit.Before
+import org.junit.Test
+import org.junit.runner.RunWith
+import java.text.SimpleDateFormat
+import java.time.LocalDate
+import java.time.format.DateTimeFormatter
+import java.util.*
+import kotlin.math.absoluteValue
+
+
+
+@RunWith(AndroidJUnit4::class)
+@LargeTest
+@BrowserstackTests
+class DateWidgetsTests: BaseTest() {
+    companion object {
+        const val CCZ_NAME = "date_widgets_tests.ccz"
+        const val APP_NAME = "Date Widgets"
+        val listOfMonths = listOf(listOf("Baishakh","Jestha","Ashadh","Shrawan",
+                            "Bhadra","Ashwin","Kartik","Mangsir","Poush","Magh",
+                            "Falgun","Chaitra"),
+            listOf("Säne","Hämle","Nähäse","P’agume",
+                "Mäskäräm","T’ïk’ïmt","Hïdar","Tahsas","T’ïr","Yäkatit",
+                "Mägabit","Miyaziya"))
+
+    }
+
+    @Before
+    fun setup() {
+        installApp(APP_NAME, CCZ_NAME)
+        InstrumentationUtility.login("test1", "123")
+    }
+
+    /**
+     * Adding the teardown method so the tests doesnot fail after only thefirst execution, when executed as a whole.
+     */
+
+    @Test
+    fun testDateWidgets(){
+        for (list in listOfMonths){
+            InstrumentationUtility.openModule(listOfMonths.indexOf(list))
+            if (listOfMonths.indexOf(list) == 0){
+                onView(withSubstring("This form will test the Nepal date widget.")).isPresent()
+            }else{
+                onView(withSubstring("This form will test the Ethiopian date widget.")).isPresent()
+            }
+
+            InstrumentationUtility.nextPage()
+            InstrumentationUtility.nextPage()
+
+            // reads the current month, gets the current index in the list, rotates the list starting from current month
+            val month_text = InstrumentationUtility.getText(onView(withId(R.id.monthtxt)))
+            list.toMutableList()
+            val index = list.indexOf(month_text)
+            Collections.rotate(list,-index)
+
+            //asserts if months are present in order
+            for (listItem in list){
+                assertTrue(onView(withText(listItem)).isPresent())
+                onView(withId(R.id.monthupbtn)).perform(ViewActions.click())
+            }
+
+            onView(withId(R.id.yeardownbtn)).perform(ViewActions.click())
+            val date_selected = setDateToUniversalCalender(-10)
```

Comment:
local vars and values should be camel case - `dateSelected` etc.

Reply:
Done!